src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/PlatformTargetBuildPropertyPageEnumProvider.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.Build.Framework.XamlTypes;
+using Microsoft.VisualStudio.Threading;
+using EnumCollection = System.Collections.Generic.ICollection<Microsoft.VisualStudio.ProjectSystem.Properties.IEnumValue>;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties.Package
+{
+    /// <summary>
+    ///     Responsible for producing valid values for the TargetFramework property from a design-time build.
+    /// </summary>
+    [ExportDynamicEnumValuesProvider("PlatformTargetEnumProvider")]
+    [AppliesTo(ProjectCapability.DotNet)]
+    internal class PlatformTargetBuildPropertyPageEnumProvider : IDynamicEnumValuesProvider, IDynamicEnumValuesGenerator
+    {
+        private readonly ProjectProperties _properties;
+
+        [ImportingConstructor]
+        public PlatformTargetBuildPropertyPageEnumProvider(ProjectProperties properties)
+        {
+            _properties = properties;
+        }
+
+        public bool AllowCustomValues => throw new NotImplementedException();
+
+        public async Task<EnumCollection> GetListedValuesAsync()
+        {
+            var result = new List<IEnumValue>();
+
+            ConfigurationGeneral configuration = await _properties.GetConfigurationGeneralPropertiesAsync();
+
+            string availablePlatformsTargets = await configuration.AvailablePlatforms.GetDisplayValueAsync();
+
+            foreach (string platformTarget in availablePlatformsTargets.Split(','))
+            {
+                result.Add(new PageEnumValue(new EnumValue() { Name = platformTarget }));
```

Comment:
You will also need to have a `DisplayName` here, same value as `platformTarget`. For the case where `platformTarget` is "AnyCPU", `DisplayName` should be "Any CPU" (with a space).

Reply:
Done